platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedState.java

```diff
@@ -487,21 +488,22 @@ public long getMinRoundGeneration() {
     }
 
     /**
-     * Check if this is a state that needs to be eventually written to disk.
+     * Set the reason why this state needs to be written to disk.
      *
-     * @return true if this state eventually needs to be written to disk
+     * @param reason the reason why this state needs to be written to disk
      */
-    public boolean isStateToSave() {
-        return stateToSave;
```

Comment:
I recommend keeping this method. You can just do `return reason != null`.

Reply:
done